langgraph_bot/tools/tools.py

```
@@ -27,3 +35,131 @@ def title_tool(posttitle:str,postdata:str):
+    """
+    NAME : PDFREADER_TOOL
+    WORK : It is used to read from the pdfs. it loads the data from the pdfs and you can read from it for better understanding.
+           it will be mostly used to read some newly published articals or the existing ones from gathering necessory infromation for post generation.
+
+    """
+    for values in list:
+        pdf_url = values.get("pdf_url")
+        if "/abs/" in pdf_url:
+            pdf_url = pdf_url.replace("/abs/", "/pdf/") + ".pdf"
+
+        response = requests.get(pdf_url)
```

Ruff (0.14.10)

Probable use of `requests` call without timeout

(S113)